src/framework/mpas_string_utils.F

```diff
@@ -0,0 +1,54 @@
+! Copyright (c) 2023 The University Corporation for Atmospheric Research (UCAR).
+!
+! Unless noted otherwise source code is licensed under the BSD license.
+! Additional copyright and license information can be found in the LICENSE file
+! distributed with this code, or at https://mpas-dev.github.io/license.html .
+!
+!-----------------------------------------------------------------------
+!  mpas_string_utils
+!
+!> \brief Collection of functions used for string manipulation
+!> \author Matthew Dimond
+!> \date   25 July 2023
+!> \details
+!>  This module provides functions and subroutines used for string
+!>  manipulations and utilities.
+!
+!-----------------------------------------------------------------------
+module mpas_string_utils
+
+  contains
+
+  subroutine mpas_split_string(string, delimiter, subStrings)
```

Comment:
I suppose we might as well add a documentation block for this subroutine now that we're migrating it into a new module. If you'd like to chat at all about how we describe this routine, just let me know.